src/pages/user-descriptor/weight-area-chart.jsx

```diff
@@ -1,50 +1,71 @@
-
-// Expected Object
-function Datapoint(uunit, uvalue, ulat, ulon, utime, ugender) {
-  this.uunit = uunit;
-  this.uvalue = uvalue;
-  this.ulat = ulat;
-  this.ulon = ulon;
-  this.utime = utime;
-  this.ugender = ugender;
+import { gql } from 'apollo-boost';
+import { useQuery } from '@apollo/react-hooks';
+
+//GraphQL ----------------------------------------------------------------
+const GET_POUNDS = gql`
+      {
+        getPaginatedDescriptors(unit: "lb", start: 0, count: 500)
+        {value}
+      }
+      `;
+
+function POUNDS() {
+  const { loading, error, data } = useQuery(GET_POUNDS);
+
+  if (loading) return 'Loading...';
+  if (error) return `Error! ${error.message}`;
+
+  return (
+      data.getPaginatedDescriptors
+  );
 }
-
-// Random Data and Object Generation function
-function GenRandomData() {
-  let arrData = [];
-  for (let i = 0; i < 200; i++) {
-    let newValue = Math.floor(Math.random() * (300 - 61 + 1) + 61);
-    let newUnit = "lb";
-    let newLat = 0;
-    let newLon = 0;
-    let newTime = 0;
-    let randGen = Math.floor(Math.random() * 2);
-    let newGender = "";
-
-    if (randGen) {
-      newGender = "Male";
-    } else {
-      newGender = "Female";
-    }
-
-    let newDatapoint = new Datapoint(
-      newUnit,
-      newValue,
-      newLat,
-      newLon,
-      newTime,
-      newGender
-    );
-    arrData.push(newDatapoint);
+//------------------------------------------------------------------------------
+function rangeIncrement(valuew, rangearray) {
+  let indx;
+  if (valuew >= 61 && valuew <= 80) {
+    rangearray[0] = rangearray[0] + 1;
+    indx = 0;
+  } else if (valuew >= 81 && valuew <= 100) {
+    rangearray[1] = rangearray[1] + 1;
+    indx = 1;
+  } else if (valuew >= 101 && valuew <= 120) {
+    rangearray[2] = rangearray[2] + 1;
+    indx = 2;
+  } else if (valuew >= 121 && valuew <= 140) {
+    rangearray[3] = rangearray[3] + 1;
+    indx = 3;
+  } else if (valuew >= 141 && valuew <= 160) {
+    rangearray[4] = rangearray[4] + 1;
+    indx = 4;
+  } else if (valuew >= 161 && valuew <= 180) {
+    rangearray[5] = rangearray[5] + 1;
+    indx = 5;
+  } else if (valuew >= 181 && valuew <= 200) {
+    rangearray[6] = rangearray[6] + 1;
+    indx = 6;
+  } else if (valuew >= 201 && valuew <= 220) {
+    rangearray[7] = rangearray[7] + 1;
+    indx = 7;
+  } else if (valuew >= 221 && valuew <= 240) {
+    rangearray[8] = rangearray[8] + 1;
+    indx = 8;
+  } else if (valuew >= 241 && valuew <= 260) {
+    rangearray[9] = rangearray[9] + 1;
+    indx = 9;
+  } else if (valuew >= 261 && valuew <= 280) {
+    rangearray[10] = rangearray[10] + 1;
+    indx = 10;
+  } else if (valuew >= 281 && valuew <= 300) {
+    rangearray[11] = rangearray[11] + 1;
+    indx = 11;
+  } else {
+    indx = 100;
   }
-  return arrData;
+  return indx;
 }
 
-// Will be replaced with call
-let arrData = GenRandomData();
-
 // Process the list for all necessary info
 function processDataList(arrData) {
```

Comment:
Assumption here is that arrData is always an array but thats actually not what happens in your code.